clang/test/CodeGenSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,163 @@
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[MACRO_MACRO_SIZE]], [[MACRO_MACRO_SIZE]]* @[[MACRO_MACRO_X]]
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[MACRO_MACRO_SIZE]], [[MACRO_MACRO_SIZE]]* @[[MACRO_MACRO_Y]]
+
+        template_param<int>();
+        // CHECK: define linkonce_odr spir_func void @_Z14template_paramIiEvv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[INT1_SIZE]], [[INT1_SIZE]]* @[[INT1]]
+
+        template_param<decltype(x)>();
+        // CHECK: define internal spir_func void @_Z14template_paramIZZ4mainENKUlvE_clEvEUlvE_Evv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[LAMBDA_SIZE]], [[LAMBDA_SIZE]]* @[[LAMBDA]]
+
+        lambda_in_dependent_function<int>();
+        // CHECK: define linkonce_odr spir_func void @_Z28lambda_in_dependent_functionIiEvv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_INT_SIZE]], [[DEP_INT_SIZE]]* @[[LAMBDA_IN_DEP_INT]]
+
+        lambda_in_dependent_function<decltype(x)>();
+        // CHECK: define internal spir_func void @_Z28lambda_in_dependent_functionIZZ4mainENKUlvE_clEvEUlvE_Evv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_LAMBDA_SIZE]], [[DEP_LAMBDA_SIZE]]* @[[LAMBDA_IN_DEP_X]]
+
+        lambda_no_dep<int, double>(3, 5.5);
+        // CHECK: define linkonce_odr spir_func void @_Z13lambda_no_depIidEvT_T0_(i32 %a, double %b)
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[NO_DEP_LAMBDA_SIZE]], [[NO_DEP_LAMBDA_SIZE]]* @[[LAMBDA_NO_DEP]]
+
+        int a = 5;
+        double b = 10.7;
+        auto y = [](int a) { return a; };
+        auto z = [](double b) { return b; };
+        lambda_two_dep<decltype(y), decltype(z)>();
+        // CHECK: define internal spir_func void @_Z14lambda_two_depIZZ4mainENKUlvE_clEvEUliE_ZZ4mainENKS0_clEvEUldE_Evv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_LAMBDA1_SIZE]], [[DEP_LAMBDA1_SIZE]]* @[[LAMBDA_TWO_DEP]]
+
+        lambda_two_dep<decltype(z), decltype(y)>();
+        // CHECK: define internal spir_func void @_Z14lambda_two_depIZZ4mainENKUlvE_clEvEUldE_ZZ4mainENKS0_clEvEUliE_Evv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_LAMBDA2_SIZE]], [[DEP_LAMBDA2_SIZE]]* @[[LAMBDA_TWO_DEP2]]
+      });
+
+  kernel_single_task<class kernel2>(func<Derp>);
+  // moved check to the beginning of the file because Filecheck has trouble matching these here
+  // call spir_func void @_Z18kernel_single_taskIZ4mainE7kernel2PFPKcvEEvT0_(i8* ()* @_Z4funcI4DerpEDTu20__unique_stable_nameXsrT_3strEEEv)
+
+  auto l1 = []() { return 1; };
+  auto l2 = [](decltype(l1) *l = nullptr) { return 2; };
+  kernel_single_task<class kernel3>(l2);
+  printf(__builtin_unique_stable_name(l2));
+  // moved the check to the beginning of the file because of trouble matching them here
```

Comment:
See my comment above, I'm guessing that is why you had to mvoe this to 'above'.

Reply:
That seems to be the case, yes!